system-addon/lib/NewTabInit.jsm

```diff
@@ -0,0 +1,23 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const {utils: Cu} = Components;
+const {actionTypes: at} = Cu.import("resource://activity-stream/common/Actions.jsm", {});
+
+/**
+ * NewTabInit - A placeholder for now. This will send a copy of the state to all
+ *              newly opened tabs.
+ */
+class NewTabInit {
+  onAction(action) {
+    switch (action.type) {
+      case at.INIT:
+        dump(`\n${JSON.stringify(this.store.getState(), 0, 2)}\n`);
```

Comment:
This will go away as soon as the `MessageManager` lands. I will add a comment. I just added it for manually testing the patch